docs/device_registration.md

```diff
@@ -0,0 +1,252 @@
+# Firefox Accounts Device Registration
+
+When using Firefox Accounts
+to connect a device to Firefox Sync,
+it is possible to annotate
+the device's sessionToken
+with additional metadata
+to improve the user experience.
+Through the Device Registration API,
+a device can:
+
+* Customize the way it appears
+  in the list of connected devices,
+  by providing a display name
+  and device type.
+* Subscribe to push notification events
+  by providing a webpush subscription endpoint.
+* Allow other devices to send it commands,
+  and send commands to other devices in turn.
+
+## Basic Registration
+
+Devices can manage their details
+by `POST`ing a registration record
+to `/v1/account/device`, like so:
+
+```
+POST /v1/account/device HTTP/1.1
+Authorization: < sessionToken HAWK header >
+{
+  "name": "my custom name",
+  "type": "desktop"
+}
+------
+HTTP/1.1 200 OK
+{
+  "id": "d3fc82acca7fc8c50acc06d21babbc00",
+  "name": "my custom name",
+  "type": "desktop"
+}
+```
+
+The server allocates a unique ID for each device,
+which can be provided
+when updating an existinf device registration.
+
+Supported device type values
+include "desktop", "mobile" and "tablet".
+
+## Push Notifications
+
+Devices can receive timely updates
+on account lifecycle events
+by providing a [WebPush Subscription](https://developer.mozilla.org/en-US/docs/Web/API/Push_API)
+to which to server can publish events.
+The subscription consists of a https endpoint,
+a public key,
+and an authentication key
+which can be used to encrypt
+the push message payload.
+
+These can be provided in the device record
+either during initial registration,
+or as an update like so::
+
+```
+POST /v1/account/device HTTP/1.1
+Authorization: < sessionToken HAWK header >
+{
+  "id": "d3fc82acca7fc8c50acc06d21babbc00",
+  "pushCallback": "https://updates.push.services.mozilla.com/ggYxFjPEjSM/Cg9Et44JFpg",
+  "pushPublicKey": "jXPJHE7-n3cNZGyYBd0yz1BA0V1uLOn-QnOg4kOS1r-oHHep5lQc8KHySevTwVPmcS0oTs_MICMjoYCgA6979Hg",
+  "pushAuthKey": "Wj18qUd0YS2-9vob79WdWg"
+}
+------
+HTTP/1.1 200 OK
+{
+  "id": "d3fc82acca7fc8c50acc06d21babbc00",
+  "name": "my custom name",
+  "type": "desktop",
+  "pushCallback": "https://updates.push.services.mozilla.com/ggYxFjPEjSM/Cg9Et44JFpg",
+  "pushPublicKey": "jXPJHE7-n3cNZGyYBd0yz1BA0V1uLOn-QnOg4kOS1r-oHHep5lQc8KHySevTwVPmcS0oTs_MICMjoYCgA6979Hg",
+  "pushAuthKey": "Wj18qUd0YS2-9vob79WdWg"
+}
+```
+
+Once a webpush subscription is registered,
+the device will receive push notifications
+that match the JSON schema defines in [`pushpayloads.schema.json`](pushpayloads.schema.json).
+Each notification will have a "command" field
+indicating the type of event,
+and a "data" field containing an object
+with additional event-specific data.
+
+The currently supported commands are:
+
```

Comment:
Should we also talk about the payload-less command that means "account/session verified"?